ts/components/services/ServicesSearch.tsx

```diff
@@ -0,0 +1,168 @@
+import * as pot from "italia-ts-commons/lib/pot";
+import React, { ComponentProps } from "react";
+
+import { SectionListData } from "react-native";
+import { ServicePublic } from "../../../definitions/backend/ServicePublic";
+import { ProfileState } from "../../store/reducers/profile";
+import { serviceContainsText } from "../../utils/services";
+import ServiceSectionListComponent from "./ServiceSectionListComponent";
+
+type OwnProps = {
+  sectionsState: // tslint:disable-next-line: readonly-array
+  Array<SectionListData<pot.Pot<ServicePublic, Error>>>;
+  searchText: string;
+  profile: ProfileState;
+  onRefresh: () => void;
+  navigateToServiceDetail: (service: ServicePublic) => void;
+};
+
+type Props = Pick<
+  ComponentProps<typeof ServiceSectionListComponent>,
+  "onRefresh"
+> &
+  OwnProps;
+
+type State = {
+  potFilteredServiceSectionsStates: pot.Pot<
+    // tslint:disable-next-line: readonly-array
+    Array<SectionListData<pot.Pot<ServicePublic, Error>>>,
+    Error
+  >;
+};
+
+/**
+ * Filter only the services that match the searchText.
+ */
+const generateSectionsServicesStateMatchingSearchTextArrayAsync = (
+  potServicesState: pot.Pot<
```

Comment:
you can avoid repeating the type 

```suggestion
  potServicesState: State["potFilteredServiceSectionsStates"]
```